lxd/device/gpu_physical.go

```diff
@@ -98,10 +101,192 @@ func (d *gpuPhysical) Start() (*deviceConfig.RunConfig, error) {
 	return d.startContainer()
 }
 
+// startCDIDevices starts all the devices given in a CDI specification:
+// * `unix-char` (representing the card and non-card devices)
+// * `disk` (representing the mounts)).
+func (d *gpuPhysical) startCDIDevices(configDevices *cdi.ConfigDevices, runConf *deviceConfig.RunConfig) error {
+	srcFDHandlers := make([]*os.File, 0)
+	defer func() {
+		for _, f := range srcFDHandlers {
+			_ = f.Close()
+		}
+	}()
+
+	for deviceName, conf := range configDevices.UnixCharDevs {
+		if conf["source"] == "" {
+			return fmt.Errorf("The source of the unix-char device %q is empty", deviceName)
+		}
+
+		_, major, minor, err := unixDeviceAttributes(conf["source"])
```

Comment:
@elezar In the CDI spec I generated, I never saw the `major` and `minor` fields being present. Is it because I need additional settings when I do the generation? This is example of a spec on a Tegra board: https://paste.ubuntu.com/p/GshmQx49Vg/

There are no mentions of `major` and `minor`

Reply:
The NVIDIA tooling does not *currently* generate a spec including major and minor numbers. The point is that we need to support the Spec in general and not the NVIDIA-specific implementation. See https://github.com/cncf-tags/container-device-interface/blob/e66544063aa7760c4ea6330ce9e6c757f8e61df2/pkg/cdi/container-edits.go#L85-L114 for how the spec edits are applied to the OCI runtime specification.

It may also be that we need to update the spec document to make this clearer.